Release machinelearningservices microsoft.machine learning services 2025 04 01

fmabroukmsft opened the pull request:
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![spec_pr_review_workflow_diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5bb5e7ce-8aff-4dbb-a3f8-0d9b68fef5b1)

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

fmabroukmsft committed:
Copy files from stable/2024-10-01

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

fmabroukmsft committed:
Update version to stable/2025-04-01

Updated the API version from stable/2024-10-01 to stable/2025-04-01.

fmabroukmsft committed:
Added tag for 2025-04-01 in readme file

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Microsoft.MachineLearningServices](https://apiview.dev/Assemblies/Review/af9cc69e6ae94f27ab2ca41fddf6ba46?revisionId=960129df5d784495b77d9085a19fa850)

ragovada commented:
/azp run

fmabroukmsft commented:
/azp run

fmabroukmsft committed:
Copy approved suppressions from previous versions (#33262)

fmabroukmsft committed:
Fix prettier failures for 2025-04-01 (#33483)

chaoyu-msft committed:
Add Registry ManagedGroupSettings to 2025-04-01 GA swagger (#33779)

* Add Registry ManagedGroupSettings to 2025-04-01 GA swagger

* fix tspconfig.yaml

* fix tspconfig.yaml

* fix tspconfig.yaml

* fix tspconfig.yaml

* remove tsg changes

Dchillakuru committed:
Adding managed network settings updates for the april version (#33722)

* adding managed network settings updates for the april version

- FirewallPublicIpAddress changes
- ManagedNetworkKind changes
- New rule status changes
- Error Information changes


---------

Co-authored-by: ZhidaLiu <zhili@microsoft.com>

fmabroukmsft committed:
Initial commit. (#34007)

fmabroukmsft commented:
/azp run

fmabroukmsft committed:
Initial commit. (#34287)

fmabroukmsft commented:
/azp run

fmabroukmsft committed:
Fix compute files formatting. (#34541)

fmabroukmsft commented:
/azp run

fmabroukmsft committed:
Run prettier on ALL files in subdirectory. (#34564)

fmabroukmsft commented:
/azp run

TimLovellSmith commented:
You need to fix avocado failures
https://github.com/Azure/azure-rest-api-specs/pull/33231/checks?check_run_id=41969602384

TimLovellSmith commented:
> You need to fix avocado failures https://github.com/Azure/azure-rest-api-specs/pull/33231/checks?check_run_id=41969602384

Never mind, its a known avocado issue

razvanbadea-msft commented:
/azp run

razvanbadea-msft commented:
I signed it of but you should check the comment about errorInformation used in example files.

fmabroukmsft commented:
/azp run SDK Validation Status

JiaqiZhang-Dev commented:
/azp run

tadelesh commented:
Go breaking changes were introduced in this PR: https://github.com/Azure/azure-rest-api-specs/pull/31240.
